clients/venice-admin-tool/src/main/java/com/linkedin/venice/KafkaTopicDumper.java

```diff
@@ -309,6 +345,79 @@ private void writeToFile(PubSubMessage<KafkaKey, KafkaMessageEnvelope, Long> rec
     }
   }
 
+  // Visible for testing
+  String getChunkMetadataLog(PubSubMessage<KafkaKey, KafkaMessageEnvelope, Long> record) throws IOException {
+    KafkaKey kafkaKey = record.getKey();
+    KafkaMessageEnvelope kafkaMessageEnvelope = record.getValue();
+    if (this.isChunkingEnabled && !kafkaKey.isControlMessage()) {
+      MessageType messageType = MessageType.valueOf(kafkaMessageEnvelope);
+      int schemaId;
+      switch (messageType) {
+        case PUT:
+          schemaId = ((Put) kafkaMessageEnvelope.payloadUnion).schemaId;
+          break;
+        case DELETE:
+          schemaId = -1;
+          break;
+        default:
+          throw new IOException(
+              "Unexpected '" + messageType + "' message from Topic: " + record.getTopicName() + " Partition: "
+                  + record.getPartition());
+      }
+
+      final ChunkKeyValueTransformer.KeyType keyType = ChunkKeyValueTransformer.getKeyType(messageType, schemaId);
+      ChunkId firstChunkId = getFirstChunkId(keyType, kafkaKey, kafkaMessageEnvelope);
+
+      if (firstChunkId == null) {
+        String chunkMetadataFormatWithoutFirstChunkMd = " ChunkMd=(type:%s)";
+        return String.format(chunkMetadataFormatWithoutFirstChunkMd, keyType);
+      } else {
+        String chunkMetadataFormatWithFirstChunkMd = " ChunkMd=(type:%s, FirstChunkMd=(guid:%s,seg:%d,seq:%d))";
+        return String.format(
+            chunkMetadataFormatWithFirstChunkMd,
+            keyType,
+            GuidUtils.getHexFromGuid(firstChunkId.producerGUID),
+            firstChunkId.segmentNumber,
+            firstChunkId.messageSequenceNumber);
+      }
+    } else {
+      return "";
+    }
+  }
+
+  private ChunkId getFirstChunkId(
+      ChunkKeyValueTransformer.KeyType keyType,
+      KafkaKey kafkaKey,
+      KafkaMessageEnvelope kafkaMessageEnvelope) {
+    final RawKeyBytesAndChunkedKeySuffix rawKeyBytesAndChunkedKeySuffix =
+        chunkKeyValueTransformer.splitChunkedKey(kafkaKey.getKey(), keyType);
+
+    final ByteBuffer chunkedKeySuffixBytes = rawKeyBytesAndChunkedKeySuffix.getChunkedKeySuffixBytes();
+    final ChunkedKeySuffix chunkedKeySuffix = chunkedKeySuffixDeserializer.deserialize(chunkedKeySuffixBytes);
+
+    switch (keyType) {
+      case WITH_VALUE_CHUNK:
+        return chunkedKeySuffix.chunkId;
+      case WITH_CHUNK_MANIFEST:
+        ChunkedValueManifest chunkedValueManifest = manifestSerializer.deserialize(
+            ByteUtils.extractByteArray(((Put) kafkaMessageEnvelope.payloadUnion).putValue),
+            AvroProtocolDefinition.CHUNKED_VALUE_MANIFEST.getCurrentProtocolVersion());
```

Comment:
I know we haven't evolve chunk value manifest yet, but it seems to be safer to use the schema id embedded in KME rather than the current version for decoding, right?